docker/base/build.sh

```diff
@@ -46,6 +46,13 @@ function extension {
   fi
 }
 
+# Detect if we are using go modules
+if [[ "$GO111MODULE" == "on" || "$GO111MODULE" == "auto" ]]; then
```

Comment:
What about `"$GO111MODULE" == ""`?  That should be the same as `auto`.

I'm also not sure about the logic here.  `GO111MODULE` alone doesn't determine whether we're using modules.  It also matters whether we're inside `GOPATH` and whether `go.mod` is present.  But maybe I'm misunderstanding the code (in which case perhaps the comment could be clearer?).

Reply:
The `GO111MODULE` gets set to `on` by `xgo` when it has detected a `go.mod` in the root of the repo. Then `build.sh` need only check for this env var and continue accordingly.